relayertest/test.go

```diff
@@ -0,0 +1,227 @@
+// Package relayertest exposes a TestRelayer function
+// that can be imported into other packages' tests.
+//
+// The exported TestRelayer function is intended to be a stable API
+// that calls the other TestRelayer_* functions, which are not guaranteed
+// to remain a stable API.
+//
+// External packages that intend to run IBC tests against their relayer implementation
+// should define their own implementation of ibc.RelayerFactory,
+// and in most cases should use an instance of ibc.BuiltinChainFactory.
+//
+//     package myrelayer_test
+//
+//     import (
+//       "testing"
+//
+//       "github.com/strangelove-ventures/ibc-test-framework/ibc"
+//       "github.com/strangelove-ventures/ibc-test-framework/relayertest"
+//     )
+//
+//     func TestMyRelayer(t *testing.T) {
+//       relayertest.TestRelayer(t, ibc.NewBuiltinChainFactory([]ibc.BuiltinChainFactoryEntry{
+//         {Name: ""},
+//       }, MyRelayerFactory())
+//     }
+//
+// Although the relayertest package is made available as a convenience for other projects,
+// the ibc-test-framework project should be considered the canonical definition of tests and configuration.
+package relayertest
+
+import (
+	"strings"
+	"testing"
+
+	transfertypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+	"github.com/strangelove-ventures/ibc-test-framework/ibc"
+	"github.com/stretchr/testify/require"
+)
+
+// TestRelayer is the stable API exposed by the relayertest package.
+// This is intended to be used by Go unit tests.
+func TestRelayer(t *testing.T, cf ibc.ChainFactory, rf ibc.RelayerFactory) {
+	t.Run("relay packet", func(t *testing.T) {
+		t.Parallel()
+
+		TestRelayer_RelayPacket(t, cf, rf)
+	})
+}
+
+func sanitizeTestNameForContainer(testName string) string {
+	return strings.ReplaceAll(testName, "/", "_")
+}
+
+// TestingT is a subset of *testing.T,
+// to support calling certain test-like functions
+// from a non-test binary.
+type TestingT interface {
```

Comment:
This interface should only be necessary if we continue in this direction to run tests in a non-test main.go. Otherwise we can fall back to a plain `*testing.T` instead.